### ISSUE TITLE
AnyTone aprsPTT setting not copied to radio

### ISSUE THREAD

**Issue by ikbenkous** [hmatuschek/qdmr#523]
Radio: `AnyTone AT-D878UVII Plus`

The channel `anytone` extension has the `aprsPTT` field that isn't copied to the radio.

I have tried with and without:
* the `anytone` extension of the APRS System added
  * the proper frequency field in there
* channel -> basic tab -> APRS drop down box set to APRS system

No combination of these things cause the `aprsPTT` field to be copied to the radio.

Open questions:
1. Is this a bug?
2. Is this unexpected behavior and can this be improved for the user?
3. Does this radio even have a _per-channel_ aprsPTT setting?

I think I have been able to set the "Upload Type" in the menu on a per-channel/VFO basis?
This setting seems to change if I change channel/VFO? Really weird.

---
As a UX aside: I have no idea how the APRS options actually transfer into my radio. qdmr in this regard is a little confusing because what controls the `#1-#8` APRS frequencies in the menu? The channels with an APRS system selected? The anytone extensions, or the primary frequencies as selected in the APRS systems itself?

What's even more confusing is that a SECOND APRS system with it's own primary frequency does not actually have that frequency copied to the radio??

Feel free to expand the scope of this issue to "Inconsistent APRS system/frequency/PTT behavior", but that may be scope creep.

**Comment by ikbenkous** [comment]
@hmatuschek Your input is greatly valued here.

**Comment by hmatuschek** [comment]
I first have to check, what has changed in the encoding of the APRS settings for the AnyTone devices. 

The translation from the qdmr APRS systems to the AnyTone APRS settings is non-trivial, as the AnyTone idea of a single APRS setting with multiple frequencies is a bit weird. For now the settings are derived from the first APRS system and the alternative frequencies are taken from the other APRS systems.

Concerning the PTT settings: They should work, unless there are changes in the codeplug. 

Thanks for checking qdmr so thoroughly.

**Comment by ikbenkous** [comment]
> For now the settings are derived from the first APRS system and the alternative frequencies are taken from the other APRS systems.

Maybe these kinds of model-specific quirks can be explained under the _Supported Radios_ tab in the about window. I'm gonna be honest, I haven't read the manual cover to cover (yet).

**Comment by hmatuschek** [comment]
> The translation from the qdmr APRS systems to the AnyTone APRS settings is non-trivial, as the AnyTone idea of a single APRS setting with multiple frequencies is a bit weird. For now the settings are derived from the first APRS system and the alternative frequencies are taken from the other APRS systems.

Obviously, I forgot how I implemented it. It is all done via the anytone extensions. Define additional frequencies within the extension of the FM APRS system and then reference the frequency within the FM channel extension. You have to define the frequency first.

**Comment by hmatuschek** [comment]
> Yeah, APRS is weirdly implemented in general... if you (or anyone else) could figure out how to implement APRS via DMR in a way that automatically switches with the selected repeater frequencies that would be really awesome.
> 
> I'll see if I can help out if I ever manage to get myself a Windows VM to do some USB tracing.

This is actually done by selecting "[Selected]" as the revert channel in the DMR APRS setting and then reference this system in the DMR channel as the position reporting system.
